### ISSUE TITLE
Schéma Itinéraires de randonnées

### ISSUE THREAD

**Issue by geoffreyaldebert** [etalab/schema.data.gouv.fr#150]
## Description du schéma

Dans le cadre du programme européen Alcotra (PITEM MITO), le Parc national des Écrins est mandaté afin de réaliser un standard d'échange de données entre les différents partenaires français et italiens. Pour cela, il a réalisé une analyse du socle commun de données permettant de définir certaines activités outdoor, dont la randonnée. Fin 2020, ce standard a été validé par les partenaires du projet Européen. La présentation de ce référentiel est accessible : https://geotrek.ecrins-parcnational.fr/ressources/presentations/2020-11-PITEM-MITO-Outdoor-Data-Presentation-socle-commun.pdf

Afin d'apporter une valeur ajoutée à ce projet, le Parc national souhaite travailler à partir de ce socle commun à la création d'un "schéma de données" concernant les itinéraires de randonnée. À terme, il est envisagé que ce schéma vienne enrichir les schémas disponibles sur le site https://schema.data.gouv.fr/, ce qui permettait de publier facilement des données standardisées et interopérables en open data, notamment sur le site data.gouv.fr.

## Stade d'avancement

Un premier atelier d'échange a été organisé le 16 mars 2021 réunissant un certain nombre d'acteurs (producteurs, réutilisateurs et experts de la donnée). Un prochain échange aura lieu pour discuter techniquement de la structure et des propriétés de ces jeux de données.

Ce schéma est :

- [x] **en investigation :** il est envisagé de créer un schéma pour modéliser une situation
- [x] **en cours d'élaboration :** ce schéma fait l'objet de discussions auprès de multiples partenaires
- [ ] **en construction :** ce schéma est en cours d'implémentation après son élaboration

**Comment by geoffreyaldebert** [comment]
J'ouvre cet issue pour avoir un espace de discussion entre les différents acteurs en amont des prochains ateliers et de la première implémentation du schéma.

**Comment by ColinMaudry** [comment]
L'URL https://geotrek.ecrins-parcnational.fr/ressources/presentations/2020-11-PITEM-MITO-Outdoor-Data-Presentation-socle-commun.pdf est cassée.

**Comment by camillemonchicourt** [comment]
Bien vu. Suite à l'atelier d'hier, tous les documents ont été regroupés dans un dossier partagé : https://geotrek.ecrins-parcnational.fr/ressources/gt/10-schema-donnees-randonnees/

Incluant le support général de présentation de l'atelier d'hier : https://geotrek.ecrins-parcnational.fr/ressources/gt/10-schema-donnees-randonnees/2021-03-16-atelier-schema-rando.pdf

Ce premier atelier a permis de : 
- Etudier le contexte et confirmer la pertinence de la création d'un schéma pour les données de randonnées
- Faire une première analyse des ressources existantes et confirmer que les données de randonnées ont une structure commune

**Comment by camillemonchicourt** [comment]
Suite à l'atelier de mardi 16 mars 2021, j'ai commencé à travailler sur une première proposition de champs pouvant composer le schéma dans un tableau partagé : https://calc.ecrins-parcnational.fr/schema-rando 
Je suis notamment parti du socle commun que nous avions défini dans le cadre du PITEM, en l'enrichissant et en commençant à le mettre en correspondance avec ce qui existe dans IGNrando', Apidae, DATAtourisme et OSM.

N'hésitez pas à en prendre connaissance, le compléter ou à ajouter des colonnes avec vos commentaires.

J'ai aussi créé un document en ligne, permettant de partager des notes, des remarques, des questions : https://pad.ecrins-parcnational.fr/p/schema-rando
N'hésitez pas à le compléter aussi (après avoir renseigné votre nom en haut à droite).

Nous vous ferons parvenir prochainement un CR du premier atelier, ainsi que des propositions de date pour travailler sur les champs du schéma.

**Comment by IdrissaD** [comment]
Bonjour, à partir des documents mis en ligne par Camille, j'ai créé une V0 du schéma au format Table Schema, il est disponible sur le GitHub des SI des Parcs nationaux : https://github.com/PnX-SI/schema_randonnee
J'ai dû faire des choix pour certains champs (difficulte, medias, geometrie ...) afin d'avoir une version complète et de valider un fichier exemple, mais ils sont bien sûr débattables.
Le choix du format Table Schema s'est fait selon une vision tabulaire des données de randonnées (telles que le tableau partagé de Camille), mais peut-être que le format JSON Schema serait plus adapté.

**Comment by IdrissaD** [comment]
Bonjour, j'ai publié une nouvelle version du schéma, les principaux changements sont la modification du format ou du contenu de quelques champs (balisage, sources, geometrie...), l'ajout de type_sol et instructions, ainsi que le dépôt de scripts et vue SQL permettant l'export des données de l'application Geotrek (utilisée entre autres par les Ecrins et les Cévennes) au format compatible avec le schéma.

**Comment by johanricher** [comment]
A mon avis utiliser des champs imbriqués n'est pas une bonne pratique. On peut voir dans le fichier exemple que cela nécessite une grammaire machine à l'intérieur même des celulles :

![image](https://user-images.githubusercontent.com/10212787/116068158-359b4d00-a68a-11eb-8f35-c362500749e8.png)

Autant cela peut être justifié pour des valeurs comme lat/long ou géométrie qui ont vocation à être uniquement produits et interprétés par une machine, le reste du temps cela doit être selon moi évité.

Il faut privilégier des champs distincts, par exemple `structure_nom`, `structure_email`.

Par ailleurs l'utilité d'identifier à la fois `structure` (producteur du fichier) et `sources` ("sources originelles de la randonnée" ?) me paraît peu évident.

**Comment by IdrissaD** [comment]
C'est tout à fait vrai pour la structure, cependant pour les sources, medias et autres champs sur ce modèle, je suis arrivé à cette solution car une randonnée peut avoir plusieurs sources, plusieurs medias, etc.
En essayant d'éviter ce JSON interne aux champs, cela donnerait pour les sources :
```nom_sources : "CC Gorges Causses Cévennes, CC Causses Aigoual"
contact_sources : "contact@gorgescausses.fr, contact@caussesaigoual.fr"```
Au risque de devoir réappairer les valeurs du champ `contact` avec leurs correspondantes du champ `nom`, j'ai pensé que cela serait une plus grande source d'erreurs que des champs "array".

Pour la distinction structure/sources : dans le cadre des randonnées, les Communautés de communes sont souvent la source primaire de la donnée car elles ont la compétence légale de gestion des sentiers et des itinéraires y passant (création, entretien, sécurisation, etc). Elles sont donc citées comme la source sur les interfaces web des parcs naturels, départements ou offices de tourisme, qui en assurent généralement la valorisation mais pas l'entretien. Ici le champ `structure` identifie la structure productrice du jeu de données dans son ensemble, celle qui l'a publié en opendata par exemple, et a pu ajouter des informations aux données transmises par la (communauté de communes) source.
C'était pour nous à la fois une question de crédit ("propriété intellectuelle") à donner aux différentes structures productrices de la donnée, et un moyen d'identifier l'interlocuteur à privilégier selon la question ou le problème à faire remonter (schématiquement : la randonnée passe par un chemin privé qui est fermé au public => source ; les géométries sont cassées => structure)
Le nom et/ou la description du champ gagneraient sans doute à être clarifiés !

**Comment by johanricher** [comment]
Pour la personne qui va utiliser le fichier publié, le seul besoin que je vois est de pouvoir contacter la personne ou organisation qui est à l'origine de l'itinéraire de randonnée, par exemple pour demander des informations complémentaires sur cet itinéraire ou pour signaler une erreur. Cette "source primaire" comme vous dite correspond en vérité à la notion de producteur de la donnée. Il faut en effet distinguer le _producteur_ de la personne ou organisation qui _publie_ le fichier. Celle-ci est est directement identifiée sur le site où le fichier est disponible au téléchargement, c'est ce qu'on appelle une métadonnée. Par conséquent, il ne paraît pas utile de l'ajouter dans les données elles-mêmes.

Par ailleurs, indiquer un autre point de contact dans le fichier risque de rendre cela confus pour l'utilisateur qui ne saura pas qui contacter.

**Comment by IdrissaD** [comment]
Effectivement, je me range à votre avis ! Je le note pour intégration au prochain groupe de modifications.

**Comment by johanricher** [comment]
Avec plaisir et bravo pour votre travail !

A l'avenir je passerai par des tickets [directement sur le dépôt du schéma](https://github.com/PnX-SI/schema_randonnee/issues) pour faire mes remarques et j'invite d'autres personnes souhaitant contribuer à faire de même.
